services/src/main/java/org/keycloak/services/managers/RealmManager.java

```diff
@@ -178,7 +179,8 @@ protected void setupAdminConsole(RealmModel realm) {
         adminConsole.setName("${client_" + Constants.ADMIN_CONSOLE_CLIENT_ID + "}");
 
         adminConsole.setRootUrl(Constants.AUTH_ADMIN_URL_PROP);
-        String baseUrl = "/admin/" + realm.getName() + "/console/";
+
+        String baseUrl = UriBuilder.fromPath("/admin/{realmName}/console/").build(realm.getName()).toString();
```

Comment:
Don't we have some kind of URL builder or constant for this? E.g.

```suggestion
        String baseUrl = session.getContext().getUri(UrlType.ADMIN).build(realm.getName()).toString();
```

Reply:
@jonkoops This creation method is sometimes executed with no valid resteasy context (for example initial creation for the master realm at first boot). If you want I can do something simpler and more similar to what we had before:

```java
String baseUrl = "/admin/" + Encode.encodePathSegment(realm.getName()) + "/console/";
```

WDYT?

Reply:
Sounds fine to me, we should find a long term solution at some point to centralize this kind of logic.